genie-web/src/main/java/com/netflix/genie/web/services/impl/JobSpecificationServiceImpl.java

```diff
@@ -520,4 +529,26 @@ private String tagsToString(final Set<String> tags) {
         // Escape quotes
         return StringUtils.replaceAll(StringUtils.replaceAll(joinedString, "\'", "\\\'"), "\"", "\\\"");
     }
+
+    /**
+     * Helper to convert archive location prefix to an archive location.
+     *
+     * @param requestedArchiveLocationPrefix archive location prefix uri
+     * @param jobId                    job id
+     * @return archive location for the job
+     */
+    private String toArchiveLocation(
+        @NotNull final String requestedArchiveLocationPrefix,
+        @NotNull final String jobId
+    ) {
+
+        String archiveLocationPrefix = requestedArchiveLocationPrefix.trim();
+
+        //Remove the last File.seperator if the user supplied it
+        if (archiveLocationPrefix.endsWith(File.separator)) {
```

Comment:
This seems unnecessary. If the user supplied a `/` at the end why strip it off only to add it again in the `return` statement.

Just have your `if` append the `jobId` if it's there else append both.